js/private/image/index.ts

```diff
@@ -28,15 +29,46 @@ type Entries = { [path: string]: Entry }
 
 type Compression = 'gzip' | 'none'
 
-function findKeyByValue(entries: Entries, value: string): string {
+function findKeyByValue(entries: Entries, value: string): string | undefined {
     for (const [key, { dest: val }] of Object.entries(entries)) {
         if (val == value) {
             return key
         }
     }
-    throw new Error(
-        `couldn't map ${value} to a path. please file a bug at https://github.com/aspect-build/rules_js/issues/new/choose`
-    )
+    return undefined
+}
+
+async function isSymbolicLink(p: string) {
+    try {
+        const firstHop = await readlink(p)
+        // bazel symlinks files from execroot to sandbox execroot hence it's expected to
+        // have a one level symlink.
+        if (firstHop != p) {
+            // if after one hop we are still in the execroot then we are on RBE or some bazel spawn
+            // strategy that doesn't use symlinks.
+            if (firstHop.startsWith(process.env.JS_BINARY__EXECROOT!)) {
+                return firstHop
+            }
+            try {
+                const secondHop = await readlink(firstHop)
+                // if the first hop is not equal to second hop then it means
+                // this is symlink generated by ctx.actions.symlink
+                if (secondHop != firstHop) {
+                    return secondHop
+                }
+            } catch (e) {
```

Comment:
The outer catch does the same thing and I think we can just drop this `catch` block?

Reply:
right...